boefjes/boefjes/storage/interfaces.py

```diff
@@ -56,8 +56,17 @@ def __init__(self, plugin_id: str):
 
 
 class DuplicatePlugin(NotAllowed):
```

Comment:
Imho this is too complex for an exception class. A few points to consider:
- The user would need to know specific field values (which could be simplified if typed as a `Literal` with choices to reduce guessing)
- The `error_message` parameter is ignored unless `field` is passed, which might be unexpected behavior
- This was my confusion about the use of the `error_message` since I expected that to be an error message that I can assign to the exception. But the use of `error_message` as a proxy for another exception's message feels unusual. Usually if an exception is based on another you’d pass the exception object itself

Perhaps to make it simpler, introduce `DuplicatePluginID` and `DuplicatePluginName` subclasses with a (default) error message?
```python3
class DuplicatePluginID(DuplicatePlugin):
    def __init__(self, plugin_id: str):
        super().__init__(f"Duplicate plugin ID: {plugin_id}")

class DuplicatePluginName(DuplicatePlugin):
    def __init__(self, plugin_name: str):
        super().__init__(f"Duplicate plugin name: {plugin_name}")
```

Reply:
@ammar92 I updated the logic so that it should work out of the box for any field